proto/minder/v1/minder.proto

```diff
@@ -2163,45 +2291,102 @@ message RuleType {
-                    optional string mode = 3;
+                    optional string mode = 3 [
+                        (buf.validate.field).string = {
+                            pattern: "^\\d+$",
+                            max_len: 6,
+                        },
+                        (buf.validate.field).ignore = IGNORE_IF_DEFAULT_VALUE
+                    ];
                 }
 
                 message ActionsReplaceTagsWithSha {
                     // List of actions to exclude from the replacement
-                    repeated string exclude = 1;
+                    repeated string exclude = 1 [
+                        (buf.validate.field).repeated = {
+                            items: {
+                                string: {
+                                    pattern: "^\\.?(?:[a-zA-Z0-9_.-]+\\/)*[a-zA-Z0-9_.-]+(?:\\.[a-zA-Z0-9]+)?$",
+                                    max_len: 200,
+                                }
+                            }
+                        },
+                        (buf.validate.field).ignore = IGNORE_IF_DEFAULT_VALUE
+                    ];
                 }
 
                 // the title of the PR
-                string title = 1;
+                string title = 1 [
+                    (buf.validate.field).string = {
+                        pattern: "[A-Za-z][-/.!?,:;[:word:] ]*",
+                        min_len: 1,
+                        max_len: 75,
+                    }
+                ];
                 // the body of the PR
-                string body = 2;
+                string body = 2 [
+                    (buf.validate.field).string = {
+                        pattern: "[A-Za-z][-/.!?,:;[:word:] ]*",
+                        min_len: 1,
+                        max_len: 65536,
+                    }
+                ];
                 repeated Content contents = 3;
                 // the method to use to create the PR. For now, these are supported:
                 // -- minder.content - ensures that the content of the file is exactly as specified
                 //                     refer to the Content message for more details
                 // -- minder.actions.replace_tags_with_sha - finds any github actions within a workflow
                 //                                           file and replaces the tag with the SHA
                 // -- minder.yq.evaluate - evaluates a yq expression on a file
-                string method = 4;
+                string method = 4 [
+                    (buf.validate.field).string = {
+                        pattern: "^minder(?:\\.[a-zA-Z0-9_]+)*$",
+                        max_len: 200,
+                    }
+                ];
                 // params are unstructured parameters passed to the method. These are optional
                 // and evaluated by the method.
                 google.protobuf.Struct params = 6;
```

Comment:
Hmm, this is tricky, because it's arbitrary content, but in a structured format that may be interpreted by the rest of our rules.

I don't think we can do much here now, but we'll get a fun bug at some point later.